### ISSUE TITLE
Feature request: --env-dir

### ISSUE THREAD

**Issue by ramnes** [docker/cli#1510]
Would there be any interest in a `--env-dir` feature, to support [envdir](https://envdir.readthedocs.io/en/latest/)-like directories?

`--env-file` is great also but implies a migration for people that use `envdir` for quite some time.

I'd be glad to work on a MR if this is something that could be merged upstream.

**Comment by ramnes** [comment]
Ouch, reading all those comments break my heart. Seems like @mattrobenolt spent quite some time on this before seeing the PR closed.

Honestly I do not particularly like the "envdir spec", if we can call it that way, but I do like retrocompatibilty when it doesn't cost much, and I feel that's the case here.

But yeah, I'm not sure if I should keep this issue open if I'm the only one expressing interest in 3 years. Maintainers, feel free to decide!

**Comment by thaJeztah** [comment]
> Ouch, reading all those comments break my heart. Seems like @mattrobenolt spent quite some time on this before seeing the PR closed.

Yes, having to close a PR, isn't an easy decision 😞 , and not something that's taken lightly by maintainers. There's a saying: "_no_ is temporary, _yes_ is forever"; once a feature is implemented, it's there to stay, and will have to be maintained / supported. Sometimes, if there are doubts about a feature, "no" is the best choice; that doesn't mean that "no" is a final answer, just that _at that point_, it was; but can be revisited in future.

> But yeah, I'm not sure if I should keep this issue open if I'm the only one expressing interest in 3 years. Maintainers, feel free to decide!

Let me keep this ticket open for now; perhaps there's more people finding it; having more information about use cases for a feature can help making these decisions. 👍
